cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -571,3 +589,252 @@ def _circuit_diagram_info_(
         SingleQubitCliffordGate.Z: SingleQubitCliffordGate.Z_nsqrt,
     },
 }
+
+
+class CommonCliffordGateMetaClass(value.ABCMetaImplementAnyOneOf):
+    """A metaclass used to lazy initialize several common Clifford Gate as class attributes."""
+
+    @property
+    def X(cls):
+        if getattr(cls, '_X', None) is None:
+            cls._Z = cls._generate_clifford_from_known_gate(1, pauli_gates.X)
+        return cls._Z
+
+    @property
+    def Y(cls):
+        if getattr(cls, '_X', None) is None:
+            cls._Z = cls._generate_clifford_from_known_gate(1, pauli_gates.Y)
+        return cls._Z
+
+    @property
+    def Z(cls):
+        if getattr(cls, '_X', None) is None:
+            cls._Z = cls._generate_clifford_from_known_gate(1, pauli_gates.Z)
+        return cls._Z
+
+    @property
+    def H(cls):
+        if getattr(cls, '_H', None) is None:
+            cls._H = cls._generate_clifford_from_known_gate(1, common_gates.H)
+        return cls._H
+
+    @property
+    def S(cls):
+        if getattr(cls, '_S', None) is None:
+            cls._S = cls._generate_clifford_from_known_gate(1, common_gates.S)
+        return cls._S
+
+    @property
+    def CNOT(cls):
+        if getattr(cls, '_CNOT', None) is None:
+            cls._CNOT = cls._generate_clifford_from_known_gate(2, common_gates.CNOT)
+        return cls._CNOT
+
+    @property
+    def CZ(cls):
+        if getattr(cls, '_CZ', None) is None:
+            cls._CZ = cls._generate_clifford_from_known_gate(2, common_gates.CZ)
+        return cls._CZ
+
+    @property
+    def SWAP(cls):
+        if getattr(cls, '_SWAP', None) is None:
+            cls._SWAP = cls._generate_clifford_from_known_gate(2, common_gates.SWAP)
+        return cls._SWAP
+
+
+class CommonCliffordGates(metaclass=CommonCliffordGateMetaClass):
```

Comment:
Why does `CommonCliffordGates` need to inherit from `CommonCliffordGateMetaClass` ? Can we not just put all of the gate property definitions from `CommonCliffordGateMetaClass` in here ?

Reply:
It was mainly because I encountered the error when adding the gate property directly. (Note they are class member instead of instance member). The problem was `raw_types.Gate` inherit the `value.ABCMetaImplementAnyOneOf` as metaclass, which is not the standard metaclass. Python required the metaclass is compatible(or maybe?) So I have to create this redundant form. Maybe there is better solution? But I am not sure how to workaround the metaclass problem here.

Reply:
Ahhh I see. In that case we can stick with it for now.